### PR TITLE
fix: cast to char* instead of cast from char*

### LINKED ISSUE #73: error: cast [...] increases required alignment of target type on riscv64
gpui-0.2.46 and gpui-0.2.47 fail to compile on riscv64. I get the following error:

```
In file included from /usr/src/RPM/BUILD/gpui-0.2.47/src/plugins/pol/binary.h:29,
                 from /usr/src/RPM/BUILD/gpui-0.2.47/src/plugins/pol/binary.cpp:22:
/usr/src/RPM/BUILD/gpui-0.2.47/src/plugins/pol/encoding.h: In instantiation of 'std::__cxx11::basic_string<_CharT> pol::convert(string_const_iterator<source_char>, string_const_iterator<source_char>, iconv_t) [with target_char = char16_t; source_char = char; string_const_iterator<source_char> = std::__cxx11::basic_string<char>::const_iterator; iconv_t = void*]':
/usr/src/RPM/BUILD/gpui-0.2.47/src/plugins/pol/encoding.h:179:45:   required from 'std::__cxx11::basic_string<_CharT> pol::convert(const std::__cxx11::basic_string<source_char, std::char_traits<_CharT>, std::allocator<_T2> >&, iconv_t) [with target_char = char16_t; source_char = char; iconv_t = void*]'
/usr/src/RPM/BUILD/gpui-0.2.47/src/plugins/pol/binary.cpp:76:68:   required from here
/usr/src/RPM/BUILD/gpui-0.2.47/src/plugins/pol/encoding.h:163:23: error: cast from 'std::array<char, 512>::pointer' {aka 'char*'} to 'char16_t*' increases required alignment of target type [-Werror=cast-align]
  163 |         result.append(reinterpret_cast<target_char *>(temp->data()),
      |                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/src/RPM/BUILD/gpui-0.2.47/src/plugins/pol/encoding.h:164:23: error: cast from 'char*' to 'char16_t*' increases required alignment of target type [-Werror=cast-align]
  164 |                       reinterpret_cast<target_char *>(outbuf));
      |                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
gmake[2]: *** [src/plugins/pol/CMakeFiles/pol-plugin.dir/build.make:76: src/plugins/pol/CMakeFiles/pol-plugin.dir/binary.cpp.o] Error 1
```

### PR DESCRIPTION
**Pull request by kiper220-alt**
fixes #73

### CODE REVIEW

File: src/plugins/pol/encoding.h [hunk 1]
```diff
@@ -149,19 +149,18 @@ inline std::basic_string<target_char> convert(string_const_iterator<source_char>
     char *inbuf = reinterpret_cast<char *>(const_cast<source_char *>(&*begin));
     size_t inbytesLeft = std::distance(begin, end) * sizeof(source_char);
 
-    auto temp = std::make_unique<std::array<char, 512>>();
-    char *outbuf = temp->data();
+    auto temp = std::make_unique<std::array<target_char, 512>>();
```
Comment (august-alt) [inline src/plugins/pol/encoding.h:152]: What is `target_char` here where it is defined?

Comment (kiper220-alt) [reply src/plugins/pol/encoding.h:152]: this is a template variable declared in "src/plugins/pol/encoding.h:143:20". "target_char" is the character type "std::basic_string<target_char>", which will be converted from "std::basic_string<source_char>".

**Comment by iv-m** [comment]
Looks nice, but unfortunately it's not enough for successful riscv64 build. 

```
[ 35%] Building CXX object src/plugins/pol/CMakeFiles/pol-plugin.dir/polplugin.cpp.o
/home/user/tests/gpui/src/plugins/pol/parser.cpp: In member function ‘void pol::PRegParser::parseHeader(std::istream&)’:
/home/user/tests/gpui/src/plugins/pol/parser.cpp:89:30: error: cast from ‘char*’ to ‘uint64_t*’ {aka ‘long unsigned int*’} increases required alignment of target type [-Werror=cast-align]
   89 |     const uint64_t header = *reinterpret_cast<uint64_t *>(&buffer[0]);
      |                              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/user/tests/gpui/src/plugins/pol/parser.cpp:90:37: error: cast from ‘const char*’ to ‘const uint64_t*’ {aka ‘const long unsigned int*’} increases required alignment of target type [-Werror=cast-align]
   90 |     const uint64_t normal_header = *reinterpret_cast<const uint64_t *>(&valid_header[0]);
      |                                     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

**Comment by iv-m** [comment]
> Looks nice, but unfortunately it's not enough for successful riscv64 build.

Ok, it's separate issue. Should I file another bug report?

BTW this time buffer is allocated on stack, which AFAIR means it's not guaranteed to be properly aligned.

**Comment by august-alt** [comment]
> > Looks nice, but unfortunately it's not enough for successful riscv64 build.
> 
> Ok, it's separate issue. Should I file another bug report?
> 
> BTW this time buffer is allocated on stack, which AFAIR means it's not guaranteed to be properly aligned.

This would be greatly appreciated. Please submit a separate issue for this bug.

**Comment by august-alt** [comment]
@iv-m I also think that a new CI pipeline (or several of them) is in order, with support for RISC-V and potentially other architectures. To prevent similar issues from ever happening in the future.

**Comment by august-alt** [comment]
@iv-m However, implementing such a pipeline will undoubtedly take time, which is why I said that the fix will probably arrive at the end of this week, if not later.

**Pull request merged**